index.js

```diff
@@ -32,14 +32,31 @@ function getDeps () {
   return readFile('package.json', 'utf8').then(function (str) {
     var json = JSON.parse(str)
     var deps = extend({}, json.dependencies)
+    var skipped = {
+      dev: 0,
+      optional: 0
+    }
     if (process.argv.indexOf('--production') === -1) {
       extend(deps, json.devDependencies)
+    } else {
+      skipped.dev = json.devDependencies ? Object.keys(json.devDependencies).length : 0
     }
     // Include optionalDependencies only if --no-optional argument is absent
     if (process.argv.indexOf('--no-optional') === -1) {
       extend(deps, json.optionalDependencies)
+    } else {
+      skipped.optional = json.optionalDependencies ? Object.keys(json.optionalDependencies).length : 0
+    }
+    var analysis_start_msg = 'Analyzing ' + Object.keys(deps).length + ' dependencies'
```

Comment:
JavaScript typically uses `lowerCamelCase`, `snake_case_like_this` is more typical for Python.